nurseconnect/utils.py

```diff
@@ -10,3 +10,72 @@ def get_period_date_format():
     datetime.date(2012, 11, 29) -> 201211
     """
     return datetime.date.today().strftime('%Y%m')
+
+
+def convert_string_to_boolean_list(string):
+    '''
+    Return a list of boolean values from a string of 'true'
+    and 'false', separated by commas.
+    '''
+    values = []
+    for answer in string.split(","):
+        ans = answer.strip().lower()
+        if(ans == "true"):
+            values.append(True)
+        elif(ans == "false"):
+            values.append(False)
+        else:
+            pass
+
+    return values
+
+
+def get_survey_results_for_user(survey, user):
+    '''
+    Return a list of survey results for a user in the following format:
+    [
+        {
+            question": question_text,
+            "user_answer": True/False,
+            "correct_answer": True/False,
+        },
+        { ... },
+    ]
+
+    Return None if the user has not submitted a survey
+
+    NOTE: This currently only works for Radio Buttons with True/False
+    and uses a hack where data stored in the survey thank you text will
+    store true, false string values seperated by commas.
+    '''
+    try:
+        submission = (survey.get_submission_class()
+                            .objects.get(page=survey, user=user))
+
+        # This is a hack, that uses answers of "true" or "false"
+        # seperated by commas in the thank you text field of a survey
+        clean_answer_key = convert_string_to_boolean_list(
+            survey.thank_you_text)
```

Comment:
Does this mean the thank you text cannot be used anymore?

Reply:
Only for surveys that are the children of articles, where we have hardcoded any relevant message that the user sees.
This will not affect surveys that are the children of `SurveysIndexPage`